File: cpp/cpp_100-999/682.cc
// C++
// leetcode 682 
// https://leetcode-cn.com/problems/baseball-game/

class Solution {
public:
    int calPoints(vector<string>& ops) {
    	int n = ops.size();
    	vector<int> tmp;
    	for(int i = 0; i < n; i++) {
    		int cur = 0;
    		if(ops[i] == "D") {
    			cur = 2 * tmp.back();
    			tmp.push_back(cur);
    		} else if(ops[i] == "+") {
    			cur = tmp.back() + tmp[tmp.size() - 2];
    			tmp.push_back(cur);
    		} else if(ops[i] == "C") {
    			tmp.pop_back();
    		} else {
    			cur = stoi(ops[i]);
    			tmp.push_back(cur);
    		}
    	}

    	int res = 0;
    	for(auto & x : tmp)
    		res += x;
    	return res;
    }
};

File: Go/Go_100-999/682.go
// Go
// leetcode 682
// https://leetcode-cn.com/problems/baseball-game/

import (
	"strconv"
)

func calPoints(ops []string) int {
	stk := make([]int, 0, 4)
	tmp := 0
	for _, s := range ops {
		switch s[0] {
		case '+':
			stk = append(stk, stk[len(stk)-1]+stk[len(stk)-2])
		case 'D':
			stk = append(stk, stk[len(stk)-1]*2)
		case 'C':
			stk = stk[0 : len(stk)-1]
		default:
			tmp, _ := strconv.Atoi(s)
			stk = append(stk, tmp)
		}
	}
	tmp = 0
	for _, x := range stk {
		tmp += x
	}
	return tmp
}